tests/test_trusted_metadata_set.py

```diff
@@ -110,13 +128,269 @@ def test_update_with_invalid_json(self):
 
             # metadata is of wrong type
             with self.assertRaises(exceptions.RepositoryError):
-                update_func(data["root"])
+                update_func(self.metadata["root"])
 
             update_func(metadata)
 
 
+    def test_verify_with_threshold_wrong_delegator(self):
+        delegated_role = Metadata.from_bytes(self.metadata["role1"])
+        timestamp = Metadata.from_bytes(self.metadata["timestamp"])
+        with self.assertRaises(ValueError):
+            verify_with_threshold(timestamp, "role1", delegated_role)
+
+    def test_verify_with_threshold_non_existent_role_name(self):
+        delegated_role = Metadata.from_bytes(self.metadata["role1"])
+        targets = Metadata.from_bytes(self.metadata["targets"])
+        with self.assertRaises(ValueError):
+            verify_with_threshold(targets, "foo", delegated_role)
+
+    def test_update_root_invalid_type(self):
+        # new_root data with invalid snapshot type
+        invalid_type_data = json.loads(self.metadata["root"])
+        invalid_type_data["signed"]["_type"] = "snapshot"
+        invalid_type_data["signed"]["meta"] = {"file1.txt": {"version": 1}}
+        invalid_type_data = json.dumps(invalid_type_data).encode()
+        # RepositoryError is thrown during new_root deserialization.
+        # It's not thrown when checking new_root.signed.type != "root"
+        with self.assertRaises(exceptions.RepositoryError):
+            self.trusted_set.update_root(invalid_type_data)
```

Comment:
I added this test because it was not tested by `test_update_with_invalid_json()`.
At least that's what coverage missing lines tells me.
Will investigate what's the reason.

Reply:
The reason was that we have forgotten to add this test there.
Will add it there with the rest of the tests testing for invalid root.

Reply:
what was the resolution here:  is this function needed or not?